### PR TITLE
build: Add server and client specific installs

### LINKED ISSUE #44: Remove requirement to install deephaven-server
**Description**

There's a couple of different ways we can use `deephaven-ipywidgets`. One way is with a local server (using a `deephaven_server.Server`), another way is using `pydeephaven` to connect to a remote client.
Currently we depend on `deephaven-server`, which causes us to download the server every time. Instead, we should remove the requirement on `deephaven-server`, and have it as an extra - `server` (and `examples`, so the examples run correctly); likewise, `pydeephaven` could be an extra install `client`.

We then need to update the README to detail how to connect using `pydeephaven`, as we currently do not include those instructions. Can use some details from the DHE instructions: https://deephaven.io/enterprise/docs/coreplus/coreplus-python-client/#jupyter
And creating a session in DHC with pydeephaven: https://deephaven.io/core/docs/tutorials/pyclient-quickstart/#1-install-pydeephaven-and-connect-to-a-running-server

### PR DESCRIPTION
**Pull request by jnumainville**
Fixes #44 
As of opening this I was able to test all the directions except for the first example in section `Displaying Tables from a Remote Server`. It should work, something is probably wrong with my setup that I need to figure out.

### CODE REVIEW

**Comment by jnumainville** [comment]
Error I was encountering should be fixed by https://github.com/deephaven/deephaven-core/issues/5848

File: README.md [hunk 1]
```diff
@@ -4,10 +4,17 @@ Deephaven Community IPython Widget Library
 
 ## Installation
 
-You can install using `pip`:
+You can install using `pip`. 
 
+If running with the embedded server, install with the following:
```
Comment (mofojed) [inline README.md:9]:
```suggestion
If running with the embedded server (`deephaven-server`), install with the following:
```

File: README.md [hunk 2]
```diff
@@ -4,10 +4,17 @@ Deephaven Community IPython Widget Library
...
+```shell
+pip install "deephaven-ipywidgets[server]"
+```
+This installs the embedded server.
+
+If connecting to a server running elsewhere, install with the following:
```
Comment (mofojed) [inline README.md:15]:
```suggestion
If connecting to a server running elsewhere (`pydeephaven`), install with the following:
```

**Review by mofojed — APPROVED** [review]
Just think we should clarify `deephaven-server` vs. `pydeephaven` right at the top. Looks great, thanks!

**Pull request merged**